### PR TITLE
Feat/pupil 680/style programme page b

### PR DESCRIPTION
**Pull request by benprotheroe**
## Description

Music year: 2011

- List of changes

add style to the programme options page 

## Issue(s)

Fixes #

## How to test

1. Go to https://deploy-preview-2425--oak-web-application.netlify.thenational.academy/pupils/beta/programmes/physics-secondary-year-11/options
2. 1. Go to https://deploy-preview-2425--oak-web-application.netlify.thenational.academy/pupils/beta/programmes/maths-secondary-year-11/options-l
3. Click on on all buttons and link
4. You should navigate to and from subject and units page

Note: there is a flicker when you go back to programmes page; theres a ticket in the backlog to fix this

### Job Story

**When** I view the ProgrammeListingPage,
**I want to see all the oak styles and components**
**So that the UX is high quality**.

### Implementation details

Create a View folder and implement the the page; using oak components and layouts; 

https://www.figma.com/file/zqpeecgzMGKRli2gUdJOLp/Pupil-Browse?type=design&node-id=4445%3A101982&mode=dev

https://www.figma.com/file/zqpeecgzMGKRli2gUdJOLp/Pupil-Browse?type=design&node-id=4515%3A114797&mode=dev

### Acceptance criteria

- [x]  page matches design
- [x] mobile, tablet, desktop
- [x] keyboard navigation
- [x] screen readable
- [x] tier and examboard logic works correctly
- [x] passes tests

### CODE REVIEW

File: src/components/PupilComponents/BrowseExamboardSelector/BrowseExamboardSelector.tsx [hunk 1]
```diff
@@ -17,41 +16,53 @@ export const BrowseExamboardSelector = ({
...
+  }
+
+  return (
+    <>
+      {" "}
+      {(() => {
+        switch (true) {
+          case !!onClick:
+            return examboards.map((examboard) => (
+              <OakPupilJourneyYearButton
```
Comment (benprotheroe) [inline src/components/PupilComponents/BrowseExamboardSelector/BrowseExamboardSelector.tsx:38]: all ive done here is swap out the oakSecondarybutton for the OakYearButton; and removed the container so the list items are rendered as the children to the programme options component

File: src/components/PupilViews/PupilLessonListing/PupilLessonListing.view.tsx [hunk 2]
```diff
@@ -41,12 +41,12 @@ export const PupilViewsLessonListing = (props: PupilLessonListingViewProps) => {
   }
 
   const breadcrumb: string[] = [yearDescription, subject];
-  if (tierDescription) {
-    breadcrumb.push(tierDescription);
-  }
```
Comment (benprotheroe) [inline src/components/PupilViews/PupilLessonListing/PupilLessonListing.view.tsx:46]: swapped this around so the breadcrumb is consistent

File: src/components/PupilViews/PupilProgrammeListing/PupilProgrammeListing.view.tsx [hunk 3]
```diff
@@ -40,101 +47,181 @@ export const PupilViewsProgrammeListing = ({
...
   const backlink = resolveOakHref({
     page: "pupil-subject-index",
     yearSlug,
   });
 
+  const subjectSlug = programmes[0]?.programmeFields.subjectSlug;
+  const subjectDescription = programmes[0]?.programmeFields.subject;
+  const yearDescriptions = programmes[0]?.programmeFields.yearDescription;
+
+  const topNavSlot = () => {
```
Comment (benprotheroe) [inline src/components/PupilViews/PupilProgrammeListing/PupilProgrammeListing.view.tsx:82]: all the changeable objects are handled by these switch functions and theyre called with the jsx

**Comment by messmike** [comment]
cheers @benprotheroe just a couple of things from me:

File: src/components/PupilComponents/BrowseTierSelector/BrowseTierSelector.tsx [hunk 4]
```diff
@@ -1,54 +1,61 @@
...
   isLegacy: boolean;
+  phaseSlug: PupilProgrammeListingData["programmeFields"]["phaseSlug"];
 }) => {
   const programmeSlugs = tiers.map(
     (tier) =>
       `${baseSlug}-${tier.tierSlug}${examboardSlug ? `-${examboardSlug}` : ""}${
         isLegacy ? "-l" : ""
       }`,
   );
+  // const phaseSlug = programmes[0]?.programmeFields.phaseSlug;
```
Comment (kimon-satan) [inline src/components/PupilComponents/BrowseTierSelector/BrowseTierSelector.tsx:32]: delete this commented out code

**Comment by RBrownsett** [comment]
@benprotheroe any chance the Choose a tier heading can be bumped to H2 rather than H1? Saves us having to H1's on the page which can be confusing for screen readers 

image

File: src/components/PupilViews/PupilProgrammeListing/PupilProgrammeListing.view.tsx [hunk 5]
```diff
@@ -40,101 +47,181 @@ export const PupilViewsProgrammeListing = ({
...
+    </OakFlex>
+  );
+
   return (
     <OakThemeProvider theme={oakDefaultTheme}>
-      <OakFlex
-        $flexDirection={"column"}
-        $gap={"space-between-m"}
-        $pa={"inner-padding-m"}
+      <AppLayout
```
Comment (kimon-satan) [inline src/components/PupilViews/PupilProgrammeListing/PupilProgrammeListing.view.tsx:194]: Could we keep this at the page level. It's confusing when we're inconsistent. Should be cut and pasteable.

File: src/components/PupilViews/PupilProgrammeListing/PupilProgrammeListing.view.tsx [hunk 6]
```diff
@@ -40,101 +47,181 @@ export const PupilViewsProgrammeListing = ({
...
+              titleSlot={
+                <OakPupilJourneyHeader
+                  iconBackground={phaseSlug}
+                  iconName={`subject-${subjectSlug}`}
+                  title={subjectDescription}
+                  breadcrumbs={breadcrumbs()}
+                />
+              }
+            >
+              {BrowseOptions()}
```
Comment (kimon-satan) [inline src/components/PupilViews/PupilProgrammeListing/PupilProgrammeListing.view.tsx:220]: This should be implemented as React Function Component
ie. <BrowseOptions/>

**Review by kimon-satan — CHANGES_REQUESTED** [review]
A few minor points.

**Review by messmike — APPROVED** [review]

**Comment by RBrownsett** [comment]
@benprotheroe might not be a problem now (because currently we don't have any exam boards with our old content) but when you have selected a subject that has both a exam board and a tier - going back to the tier page from the unit page briefly shows the exam board page. Video below:

https://github.com/oaknational/Oak-Web-Application/assets/30384846/763f4e2c-d8d5-4b34-bd80-94b7424a6d45

**Review by kimon-satan — APPROVED** [review]

**Comment by RBrownsett** [comment]
@benprotheroe noticed that the tooltip text is not fully viewable when in the mobile mode - I know we have PUPIL-701 to fix the tooltip - do we want to do something now or wait for that ticket?

image

image

**Comment by benprotheroe** [comment]
> @benprotheroe might not be a problem now (because currently we don't have any exam boards with our old content) but when you have selected a subject that has both a exam board and a tier - going back to the tier page from the unit page briefly shows the exam board page. Video below:
> 
>  Biology.secondary.Year.10.-.Unit.listing._.Oak.National.Academy.-.15.May.2024.mp4



> @benprotheroe might not be a problem now (because currently we don't have any exam boards with our old content) but when you have selected a subject that has both a exam board and a tier - going back to the tier page from the unit page briefly shows the exam board page. Video below:
> 
>  Biology.secondary.Year.10.-.Unit.listing._.Oak.National.Academy.-.15.May.2024.mp4

have made a ticket to fix this: https://www.notion.so/oaknationalacademy/24-Spring-OKRs-Sprint-7-40f22700ac7a4168ad18accaa0c99296?p=fe795b6cfe394b9a904adabbfeed9ef4&pm=s

**Comment by benprotheroe** [comment]
> @benprotheroe noticed that the tooltip text is not fully viewable when in the mobile/tablet mode - I know we have PUPIL-701 to fix the tooltip - do we want to do something now or wait for that ticket?
> 
> image
> 
> image
> 
> image

either way the fix will be on oak components repo so i will make a ticket for now and we can fix prior to launch: https://www.notion.so/oaknationalacademy/24-Spring-OKRs-Sprint-7-40f22700ac7a4168ad18accaa0c99296?p=9c312489a3d64bbfa9fa7f5c91856657&pm=s

**Pull request merged**